benchmarks/launch_latency/launch_latency.cpp

```diff
@@ -0,0 +1,259 @@
+    timer.reset();
+  }
+
+  const double x = 1.e6 / M;
+  printf("%i %i %i %i", N, V, K, M);
+  if (!opts.no_par_for) {
+    printf(" parallel_for: %lf %lf ( %lf %lf )", x * time_no_fence,
+           x * time_fence, x * time_no_fence_fenced, x * time_fence_fenced);
+  }
+  if (!opts.no_par_reduce) {
+    printf(" parallel_reduce: %lf %lf ( %lf %lf )", x * time_red_no_fence,
+           x * time_red_fence, x * time_red_no_fence_fenced,
+           x * time_red_fence_fenced);
+  }
+  if (!opts.no_par_reduce_view) {
+    printf(" parallel_reduce(view): %lf %lf ( %lf %lf )",
+           x * time_red_view_no_fence, x * time_red_view_fence,
+           x * time_red_view_no_fence_fenced, x * time_red_view_fence_fenced);
+  }
+  printf("\n");
+}
+int main(int argc, char* argv[]) {
+  Kokkos::initialize(argc, argv);
+  {
+    int N = 10000;
+    int M = 1;
+    int K = 1;
+
+    Opts opts;
+
+    printf("==========================\n");
+    printf("Kokkos Launch Latency Test\n");
+    printf("==========================\n");
+    printf("\n\n");
+    printf("Usage: %s ARGUMENTS [OPTIONS...]\n\n", argv[0]);
+    printf("Arguments: N M K\n");
+    printf("  N: loop length\n");
+    printf("  M: how many kernels to dispatch\n");
+    printf(
+        "  K: nested loop length (capped by size of functor member array\n\n");
+    printf("Options:\n");
+    printf("  --no-parallel-for:         skip parallel_for benchmark\n");
+    printf("  --no-parallel-reduce:      skip parallel_reduce benchmark\n");
+    printf(
+        "  --no-parallel-reduce-view: skip parallel_reduce into view "
+        "benchmark\n");
+    printf("\n\n");
+    printf("  Output V is the size of the functor member array\n");
+    printf("\n\n");
+
+    for (int i = 1; i < argc; ++i) {
+      const std::string_view arg(argv[i]);
+
+      // anything that doesn't start with --
+      if (arg.size() >= 2 && arg[0] != '-' && arg[1] != '-') {
+        if (i == 1)
+          N = atoi(arg.data());
+        else if (i == 2)
+          M = atoi(arg.data());
+        else if (i == 3)
+          K = atoi(arg.data());
+        else {
+          throw std::runtime_error("unexpected argument!");
+        }
+      } else if (arg == "--no-parallel-for") {
+        opts.no_par_for = true;
+      } else if (arg == "--no-parallel-reduce") {
+        opts.no_par_reduce = true;
+      } else if (arg == "--no-parallel-reduce-view") {
+        opts.no_par_reduce_view = true;
+      } else {
+        std::stringstream ss;
+        ss << "unexpected option " << arg;
+        throw std::runtime_error(ss.str());
+      }
+    }
+
+    printf(
+        "N V K M time_no_fence time_fence (time_no_fence_fenced "
+        "time_fence_fenced)\n");
+
+    run<1>(N, M, K <= 1 ? K : 1, opts);
+    run<16>(N, M, K <= 16 ? K : 16, opts);
+    run<200>(N, M, K <= 200 ? K : 200, opts);
+    run<3000>(N, M, K <= 3000 ? K : 3000, opts);
+    run<30000>(N, M, K <= 30000 ? K : 30000, opts);
```

Comment:
Do we really need/want all of these?

Reply:
@crtrott told me the motivation was that backends have different launch strategies depending on functor size, but I'm not sure why these numbers are exactly what they are. The whole thing takes a fraction of a second to run, so I don't think it's too burdensome

Reply:
OK